pkg/epp/requestcontrol/director.go

```diff
@@ -89,42 +91,59 @@ type Director struct {
 	defaultPriority int
 }
 
-// HandleRequest orchestrates the request lifecycle.
-// It always returns the requestContext even in the error case, as the request context is used in error handling.
-func (d *Director) HandleRequest(ctx context.Context, reqCtx *handlers.RequestContext) (*handlers.RequestContext, error) {
-	logger := log.FromContext(ctx)
+// getInferenceObjective fetches the inferenceObjective from the datastore otherwise creates a new one based on reqCtx.
+func (d *Director) getInferenceObjective(logger logr.Logger, reqCtx *handlers.RequestContext) *v1alpha2.InferenceObjective {
```

Comment:
can we remove logger from the arguments?
This comes back to a discussion we had on a different PR. the convention we use in the project is contextual logging, we take logger from context and use it instead of passing around logger as arg.
e.g.,
```go
log.FromContext(ctx).V(logutil.VERBOSE).Info(...)
```